packages/babel-core/src/config/option-manager.js

```diff
@@ -64,29 +65,100 @@ const optionNames = new Set([
 ]);
 
 const ALLOWED_PLUGIN_KEYS = new Set([
-  "name",
+  "after",
+  "before",
+  "capabilities",
+  "inherits",
   "manipulateOptions",
-  "pre",
+  "name",
   "post",
+  "pre",
   "visitor",
-  "inherits",
 ]);
 
 export default function manageOptions(opts: {}): {
   options: Object,
-  passes: Array<Array<[ Plugin, ?{} ]>>,
+  plugins: Array<[ Plugin, ?{} ]>,
 }|null {
   return new OptionManager().init(opts);
 }
 
+function sortPlugins(plugins) {
+  const graph = new DagMap();
+  const capabilitiesToPluginIdMap = Object.create(null);
+
+  // It is theoretically possible that something arrives at this point without
+  // having a name or key property. Make sure that it has one.
+  const unnamedPluginPrefix = "unnamedPlugin";
+  plugins.forEach((pluginTuple, index) => {
+    const plugin = pluginTuple[0];
+    if (!plugin.key) { plugin.key = `${unnamedPluginPrefix}${index}`; }
+  });
```

Comment:
just noting that we probably don't need this bit, given that we're making naming compulsory